File: Stack_Queue/function_development.py
def solution(progresses, speeds):
    answer = []   
    
    while progresses[0] < 100:
        result = 0
        
        if progresses[0] == 0:
            break
            
        for i in range(len(progresses)):
            if progresses[i] != 0:
                progresses[i] += speeds[i]   
            else:
                break

        while progresses[0] >= 100:       
            progresses.pop(0)
            speeds.pop(0)
            progresses.append(0)
            speeds.append(0)
            result += 1
        
        if result > 0:
            answer.append(result)        
                        
    return answer

'''
정확성  테스트
테스트 1 〉	통과 (0.02ms, 10.3MB)
테스트 2 〉	통과 (0.77ms, 10.2MB)
테스트 3 〉	통과 (0.71ms, 10.2MB)
테스트 4 〉	통과 (0.24ms, 10.2MB)
테스트 5 〉	통과 (0.02ms, 10.2MB)
테스트 6 〉	통과 (0.05ms, 10.2MB)
테스트 7 〉	통과 (0.96ms, 10.2MB)
테스트 8 〉	통과 (0.12ms, 10.3MB)
테스트 9 〉	통과 (0.74ms, 10.2MB)
테스트 10 〉	통과 (0.38ms, 10.2MB)
테스트 11 〉	통과 (0.02ms, 10.2MB)
채점 결과
정확성: 100.0
합계: 100.0 / 100.0
'''

File: Programmers/Stack_Queue/function_development.js
function solution(progresses, speeds) {
  var answer = [];
  let count = 0;
  
  progresses = progresses.reverse();
  speeds = speeds.reverse();
  
  while(progresses.length != 0){
      if (progresses[progresses.length-1] < 100) {
          progresses = progresses.map((value, index, array) => {return value + speeds[index]})
      } else {
          while(progresses[progresses.length-1] >= 100) {
              progresses.pop();
              speeds.pop();
              count ++;              
          }
          answer.push(count);
          count = 0;
      }
  }   
  return answer;
}

/*
array.map((value, index, array) => {return x})
value는 array에서 하나씩 가져오는 값
index는 해당 값의 인덱스
array는 배열 원형
x는 value에 새로 return해서 넣는 값
*/

/*
정확성  테스트
테스트 1 〉	통과 (0.08ms, 29.7MB)
테스트 2 〉	통과 (0.21ms, 30.1MB)
테스트 3 〉	통과 (0.28ms, 30.1MB)
테스트 4 〉	통과 (0.15ms, 29.9MB)
테스트 5 〉	통과 (0.14ms, 30.1MB)
테스트 6 〉	통과 (0.13ms, 30.1MB)
테스트 7 〉	통과 (0.25ms, 30.1MB)
테스트 8 〉	통과 (0.18ms, 29.8MB)
테스트 9 〉	통과 (0.31ms, 30.1MB)
테스트 10 〉	통과 (0.36ms, 30MB)
테스트 11 〉	통과 (0.12ms, 30.1MB)
채점 결과
정확성: 100.0
합계: 100.0 / 100.0
*/